diff --git a/cpu/cpu_nacl.go b/cpu/cpu_nacl.go
new file mode 100644
index 0000000..67e78e9
--- /dev/null
+++ b/cpu/cpu_nacl.go
@@ -0,0 +1,15 @@
+// +build nacl
+
+package cpu
+
+import (
+	"time"
+)
+
+func Times(percpu bool) ([]TimesStat, error) {
+	return []TimesStat{}, nil
+}
+
+func Percent(interval time.Duration, percpu bool) ([]float64, error) {
+	return []float64{}, nil
+}
\ No newline at end of file
diff --git a/disk/disk_nacl.go b/disk/disk_nacl.go
new file mode 100644
index 0000000..19cfd94
--- /dev/null
+++ b/disk/disk_nacl.go
@@ -0,0 +1,7 @@
+// +build nacl
+
+package disk
+
+func Usage(path string) (*UsageStat, error) {
+	return &UsageStat{}, nil
+}
diff --git a/mem/mem_nacl.go b/mem/mem_nacl.go
new file mode 100644
index 0000000..4938b9a
--- /dev/null
+++ b/mem/mem_nacl.go
@@ -0,0 +1,7 @@
+// +build nacl
+
+package mem
+
+func VirtualMemory() (*VirtualMemoryStat, error) {
+	return &VirtualMemoryStat{}, nil
+}
\ No newline at end of file
